### PR TITLE
Add a warning and build workflow section to proc-macro documentation

### PR DESCRIPTION
**Pull request by jplatte**
Follow-up to #1392, cc @bendk @Hywan.

### CODE REVIEW

File: docs/manual/src/proc_macro/index.md [hunk 1]
```diff
@@ -12,6 +12,17 @@ types in UDL must also be declared in UDL.
 
 [Procedural Macros]: https://doc.rust-lang.org/reference/procedural-macros.html
 
+**⚠ Warning ⚠** As the page title says, this is experimental. Bugs are expected and if you want to
+try it is recommended that you use `uniffi` as a git dependency so you don't run into issues that
+are already fixed.
+
+## Build workflow
+
+Before any of the things discussed below work, make your to update your bindings generation steps
```
Comment (Hywan) [inline docs/manual/src/proc_macro/index.md:21]:
```suggestion
Before any of the things discussed below work, make sure to update your bindings generation steps
```

?

File: docs/manual/src/proc_macro/index.md [hunk 2]
```diff
@@ -12,6 +12,17 @@ types in UDL must also be declared in UDL.
...
+to start with a build of your library and add
+`--lib-file <CARGO WORKSPACE>/target/<TARGET>/<BUILT CDYLIB OR STATICLIB>` to your `uniffi-bindgen`
+commandline invocation.
```
Comment (Hywan) [inline docs/manual/src/proc_macro/index.md:24]:
```suggestion
command line invocation.
```

**Review by Hywan — APPROVED** [review]
Thanks!

**Review by bendk — APPROVED** [review]

**Comment by jplatte** [comment]
Merge please? 😄

**Pull request merged**